spec/controllers/sessions_controller_spec.rb

```diff
@@ -0,0 +1,143 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe SessionsController, type: :controller do
+  describe 'GET#new' do
+    context 'when user_id session not exists' do
+      it 'returns a successful response' do
+        get :new
+
+        expect(response).to be_successful
+      end
+
+      it 'renders the template of :new action' do
+        get :new
+
+        expect(response).to render_template(:new)
+      end
+    end
+
+    context 'when user_id session exists' do
+      it 'redirects to home page' do
+        user = Fabricate(:user)
+        session[:user_id] = user[:id]
+
+        get :new
+
+        expect(response).to redirect_to(root_path)
+      end
+    end
+  end
+
+  describe 'POST#create' do
+    context 'when user_id session exists' do
+      it 'redirects to home page' do
+        user = Fabricate(:user)
+        session[:user_id] = user[:id]
```

Comment:
can use use `user.id` 🤔, same for `user.username` 🤔

Reply:
We cannot use `user.username` because `current_user` method use user_id from session to find the user in database

Reply:
Fixed in 90cafb8f77d4b8e7c78397ec65acf066b7fdc194